src/ngraph/autodiff/adjoints.cpp

```diff
@@ -125,7 +125,7 @@ std::shared_ptr<Node> autodiff::Adjoints::get(const std::shared_ptr<Node>& x)
 void autodiff::Adjoints::add_delta(const std::shared_ptr<Node>& x,
                                    const std::shared_ptr<Node>& delta)
 {
-    if (!x->has_same_type(delta))
+    if (!x->has_same_type(delta) && delta->get_shape() != x->get_outputs().at(0).get_shape())
```

Comment:
This is a bit of technical debt that doesn't matter for batchnorm but will matter for other multi-output ops. I've created NGMX-422 to track it.